### PR TITLE
Add Boxel::InputGroup and copy-to-clipboard helper

### PR DESCRIPTION
**Pull request by lukemelia**
The goal of this PR is to allow the copy to clipboard treatment. The approach I took is much more general but useful, I hope.

### CODE REVIEW

**Review by danibonilha — COMMENTED** [review]
It's looking good to me, but i'll leave to more experienced web devs to approve

File: packages/boxel/addon/components/boxel/input-group/index.css [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    );
+  --boxel-input-group-border-color: var(--boxel-purple-300);
+  --boxel-input-group-border-radius: var(--boxel-border-radius-sm);
+  /*
+  $input-group-addon-padding-y:           $input-padding-y;
+  $input-group-addon-padding-x:           $input-padding-x;
+  $input-group-addon-font-weight:         $input-font-weight;
+  $input-group-addon-color:               $input-color;
+  $input-group-addon-bg:                  $gray-200;
+  */
```
[the remark this answers is not in the document]
Comment (lukemelia) [reply packages/boxel/addon/components/boxel/input-group/index.css:16]: Yes, I will clean that up. Thanks.

**Comment by backspace** [comment]
I haven’t looked at the code yet but do you think the focus outline can be tweaked? The tiny gaps are irksome to me 😆

Boxel Components 2022-10-05 11-07-23

Bootstrap:

Input group · Bootstrap v5 2 2022-10-05 11-06-44

**Comment by lukemelia** [comment]
@backspace 
> I haven’t looked at the code yet but do you think the focus outline can be tweaked? The tiny gaps are irksome to me 😆
> hmm the preview deployment doesn’t have the copy example? I see it in the template but not rendered:

I've addressed both of those issues. Thanks for catching them.

**Review by backspace — CHANGES_REQUESTED** [review]
The implementation and interaction looks good to me, Glint feels so powerful 🤩

There are some CSS fixes I think should happen.

The outline for the error state matches but the animation when the state changes should match too; this is at 25% speed to highlight the lag:
outline-lag

Though Bootstrap doesn’t apply the error outline to the accessory at all! 🤔

Validation · Bootstrap v5 2 2022-10-05 12-49-33


It looks broken in Safari, though with focus I think it looks the best of the three browsers I checked:
safari-weirdness

Tiny gaps persist in Firefox and Chrome:
firefox-gap
chrome-gap

There’s a lot of misalignment in Safari:
more-safari

Finally, this is maybe difficult with the existing button styles but I think there should be dividers between the buttons, they look strange to me without it:

Boxel Components 2022-10-05 12-46-54

Bootstrap, for example:
Input group · Bootstrap v5 2 2022-10-05 12-47-50

**Comment by lukemelia** [comment]
I experimented with outline-offset to get tighter focus rects for this component but I observed inconsistent application of the rule between Safari and Chrome and could not get a good cross-browser effect when trying with negative values.

**Comment by lukemelia** [comment]
It turns out to be quite difficult to do the vertical border for just buttons, so instead I added a CSS Var to control the interior border width.

**Comment by lukemelia** [comment]
@backspace ready for another look

**Review by backspace — APPROVED** [review]
It all looks better! Safari has tiny gaps at some button edges but maybe we can defer that until this is actually in use:

Boxel Components 2022-10-06 09-30-48

**Comment by lukemelia** [comment]
@backspace 
>Safari has tiny gaps at some button edges

Fixed, thanks!

**Pull request merged**